Update for Stardew Valley 1.3.36 and SMAPI 3.0

Pathoschild opened the pull request:
This pull request has ongoing development since the last official update in July 2018 to maintain compatibility with the latest versions, update for new SMAPI features, and make minor improvements. Once this is merged, I can provide a compiled package to release if you'd like.

## Changes
* Updated for Stardew Valley 1.3.36 and SMAPI 3.0.
* Updated the mod build package.
* Migrated to the new simplified project format.
* Eliminated `ShopMenu2` and rewrote code to use the vanilla `ShopMenu`. This...
  * simplifies maintenance;
  * fixes controller support in shop menus;
  * fixes shop translations replaced with English;
  * fixes compatibility with other mods which handle shop menus.
* Unpacked `assets/bundles.xnb` for easier editing.
* Minor refactoring.

## Testing
The unofficial updates have almost thirteen thousand downloads combined, so the changes should be pretty stable. See download counts for the most recent releases: [unofficial.7](https://community.playstarbound.com/threads/updating-mods-for-stardew-valley-1-3.142524/page-76#post-3342338), ~~[unofficial.6](https://community.playstarbound.com/threads/updating-mods-for-stardew-valley-1-3.142524/page-76#post-3342222)~~, [unofficial.5](https://community.playstarbound.com/threads/updating-mods-for-stardew-valley-1-3.142524/page-58#post-3325754), [unofficial.3](https://community.playstarbound.com/threads/updating-mods-for-stardew-valley-1-3.142524/page-49#post-3321250), [unofficial.3](https://community.playstarbound.com/threads/updating-mods-for-stardew-valley-1-3.142524/page-49#post-3321250), [unofficial.2](https://community.playstarbound.com/threads/updating-mods-for-stardew-valley-1-3.142524/page-33#post-3315200), [unofficial.1](https://community.playstarbound.com/threads/updating-mods-for-stardew-valley-1-3.142524/page-33#post-3315199), [unofficial.1-mizzion](https://community.playstarbound.com/threads/updating-mods-for-stardew-valley-1-3.142524/page-14#post-3286214).

Let me know if you want me to make any changes!

Pathoschild committed:
remove SFarmer alias (no longer needed)

Pathoschild committed:
update for SMAPI 3.0

ToyKungZ commented on SortMyStuff/manifest.json:
```diff
@@ -5,6 +5,6 @@
-  "MinimumApiVersion": "2.8-beta",
+  "MinimumApiVersion": "2.9.0",
   "UpdateKeys": []
 }
```
"chestCoords": "73, 14",
  "chestDefs": ""
}

ToyKungZ replied:
"chestCoords": "73, 14"
  "chestDefs": "",
}

Pathoschild committed:
update mod build package

The mod build package now recognises files in an 'assets' folder as part of the mod automatically.

Pathoschild committed:
tweak solution folder convention

Pathoschild committed:
migrate to newer project format

Pathoschild committed:
update for field changes in Stardew Valley 1.4

(Also compatible with 1.3.36.)

Pathoschild committed:
reduce duplicate code

Pathoschild committed:
rewrite reopen-main-menu logic to reduce need for custom menus

Pathoschild committed:
rewrite achievements logic to reduce need for custom menus

Pathoschild committed:
move bundle logic out of custom menu

Pathoschild committed:
remove ShopMenu2, use vanilla shop menu instead

Pathoschild committed:
unpack XNB file

Pathoschild committed:
set x86 explicitly in project configuration
